### PR TITLE
Add YAML frontmatter preservation support

### PR DESCRIPTION
**Pull request by leynos**
Documents that begin with a YAML frontmatter block (starting with --- and ending with --- or ...) now have that block preserved exactly while the remainder of the document is formatted normally.

Changes:
- Add src/frontmatter.rs with split_leading_yaml_frontmatter() helper
- Wire frontmatter detection through library pipeline in src/process.rs
- Wire frontmatter protection through CLI transforms in src/main.rs
- Add comprehensive unit and behavioural CLI tests
- Update README.md with YAML frontmatter documentation
- Update docs/architecture.md with pipeline description

The frontmatter block is shielded from all transforms including --breaks, --wrap, --renumber, and table reflow.

## Summary by Sourcery

Preserve leading YAML frontmatter blocks unchanged while continuing to format the rest of Markdown documents, and document this behavior in the user and architecture docs.

New Features:
- Add YAML frontmatter detection and splitting so a leading YAML block is preserved verbatim while the remaining Markdown body is processed.
- Expose frontmatter splitting through the public library API for reuse by consumers.

Enhancements:
- Integrate frontmatter preservation into both the core processing pipeline and CLI-only transforms so options like wrapping, list renumbering, and thematic break formatting do not alter frontmatter.
- Refine documentation style and architecture docs, including fenced code/diagram formatting and citation markup, and add an execution plan document for the frontmatter feature.
- Apply minor internal refactors and style cleanups (e.g., helper inlining, clone usage) across wrapping, HTML conversion, footnotes, and tests to align with project conventions.

Documentation:
- Update README with a YAML frontmatter section including before/after examples.
- Update architecture documentation to describe the new frontmatter split step in the processing pipeline and add an execplan document outlining the feature design and implementation steps.

Tests:
- Add unit tests for frontmatter detection and preservation, including edge cases like unmatched delimiters and dotted closers.
- Add CLI integration tests covering frontmatter behavior under normal formatting, wrapping, thematic break processing, and in-place mode.
- Extend existing test suites with small adjustments to ensure new behavior is covered without regressions.

Chores:
- Switch the Rust toolchain to a specific nightly version and include rust-analyzer as a component for development tooling.
- Tidy internal contributor guidelines and documentation style guide formatting for clarity.

### CODE REVIEW

File: docs/execplans/yaml-frontmatter.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
+# Preserve leading YAML frontmatter while formatting Markdown
+
+This ExecPlan (execution plan) is a living document. The sections
+`Constraints`, `Tolerances`, `Risks`, `Progress`, `Surprises &
+Discoveries`, `Decision Log`, and `Outcomes & Retrospective` must be kept up
+to date as work proceeds.
+
+Status: DRAFT
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/execplans/yaml-frontmatter.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/process.rs [hunk 2]
```diff
@@ -339,4 +347,71 @@ mod tests {
...
+            "More text".to_string(),
+        ];
+        let out = process_stream(&input);
+        // All lines should be processed as body (no special frontmatter handling)
+        // The lines may be wrapped together, so just verify the content is present
+        assert!(out[0].contains("---"));
+        let joined = out.join("\n");
+        assert!(joined.contains("Not frontmatter"));
+        assert!(joined.contains("More text"));
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/process.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/frontmatter.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            "  This is a multi-line".to_string(),
+            "  YAML value".to_string(),
+            "---".to_string(),
+            "# Content".to_string(),
+        ];
+        let (prefix, body) = split_leading_yaml_frontmatter(&lines);
+        assert_eq!(prefix.len(), 5);
+        assert_eq!(body.len(), 1);
+        assert_eq!(body[0], "# Content");
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/frontmatter.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/process.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (leynos) [reply src/process.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/cli.rs [hunk 4]
```diff
@@ -390,3 +390,135 @@ fn test_cli_fences_null_language(#[case] input: &str, #[case] expected: &'static
...
+    let expected = concat!(
+        "---\n",
+        "title: Example\n",
+        "---\n",
+        "\n",
+        "| A | B |\n",
+        "| 1 | 2 |\n",
+    );
+    run_in_place(&[], input, expected);
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/cli.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/process.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (leynos) [reply src/process.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
